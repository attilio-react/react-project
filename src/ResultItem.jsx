import React from 'react';

import {Label} from './Label.jsx';
import {Button} from './Button.jsx';

class ResultItem extends React.PureComponent {
  render () {
    const {item} = this.props;
    return <div>
                  <p><img src={item.imgSrc} /></p>
                  <p>
		     <Label text={item.releaseDate} />
		     &nbsp;
                     <Label text={item.genre} />
		  </p>
                  <p><Label text={item.title} /></p>
                  
           </div>;
  }
}


export {SearchByContainer}


