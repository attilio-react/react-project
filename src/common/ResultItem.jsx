import React from 'react';
import {Link} from "react-router-dom";

import {paths} from 'Common/Paths.jsx';

import {Label} from 'Common/Label.jsx';
import {Button} from 'Common/Button.jsx';

class ResultItem extends React.PureComponent {

 render () {
    const {item, clickCallback} = this.props,
          self = this;
    return <div style={{border: '1px solid black'}}><Link to={paths.DETAILS_PATH + item.id}>
                  <p><img src={item.poster_path} /></p>
                  <p>
		     <Label text={item.release_date} />
		     &nbsp;
                     <Label text={item.genres.join(', ')} />
		  </p>
                  <p><Label text={item.title} /></p>
                 </Link> 
           </div>;
  }
}


export {ResultItem}


