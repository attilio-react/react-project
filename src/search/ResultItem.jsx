import React from 'react';

import {Label} from 'Common/Label.jsx';
import {Button} from 'Common/Button.jsx';

class ResultItem extends React.PureComponent {
  render () {
    const {item} = this.props;
    return <div>
                  <p><img src={item.poster_path} /></p>
                  <p>
		     <Label text={item.release_date} />
		     &nbsp;
                     <Label text={item.genres.join(', ')} />
		  </p>
                  <p><Label text={item.title} /></p>
                  
           </div>;
  }
}


export {ResultItem}


