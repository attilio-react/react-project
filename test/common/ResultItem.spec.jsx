import * as React from 'react';
import {mount} from 'enzyme';
import renderer from 'react-test-renderer';


import {ResultItem} from '|common/ResultItem.jsx';


const MOVIE_ID = 42,
      movie = {
       'id': MOVIE_ID,
       'title': 'test movie',
       'tagline': 'the best movie ever',
       'vote_average': 5.0,
       'vote_count': 99999,
       'release_date': '1970-01-01',
       'poster_path': 'http://example.com/poster.png',
       'overview': 'this is the best movie ever created',
       'budget': 9999999,
       'revenue': 999999999,
       'runtime': 123,
       'genres': ['Drama', 'Thriller', 'Comedy', 'War', 'Horror', 'All']
     }


describe('ResultItem rendering', () => {
  it('Should render result item', () => {
    const component = renderer.create(<ResultItem item={movie} />),
        json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});

describe('ResultItem behaviour', () => {
  it('Should call callback when clicked', () => {
    const onClick = jest.fn(),
      resultItem = mount(<ResultItem item={movie} clickCallback={onClick} />),
      nativeLink = resultItem.find('a').first();
    nativeLink.simulate('click');
    expect(onClick).toBeCalledWith(MOVIE_ID)
  });
});

