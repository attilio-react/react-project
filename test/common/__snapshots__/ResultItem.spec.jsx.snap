// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultItem rendering Should render result item 1`] = `
<div
  style={
    Object {
      "border": "1px solid black",
    }
  }
>
  <a
    onClick={[Function]}
  >
    <p>
      <img
        src="http://example.com/poster.png"
      />
    </p>
    <p>
      <span>
        1970-01-01
      </span>
      Â 
      <span>
        Drama, Thriller, Comedy, War, Horror, All
      </span>
    </p>
    <p>
      <span>
        test movie
      </span>
    </p>
  </a>
</div>
`;
